### ISSUE TITLE
Very FATAL BUG about hvd.broadcast_optimizer_state

### ISSUE THREAD

**Issue by yh673025667** [uber/horovod#526]
It seems that hvd.broadcast_optimizer_state **cannot broadcast optimizer state to other gpus or other machines gpus.**

i have running `pytorch_imagenet_resnet50.py`, when i load a pretrained checkpoint, it seems that only my 1st gpu on my rank0 's learning have been changed to pretrained checkpoint. but others are **still init learning!!!**
my experiments result are as follows:

=> loading checkpoint 'horovod_imagenet_v6/checkpoint-180.pth.tar'
SGD (
Parameter Group 0
    dampening: 0
    lr: 0.1
    momentum: 0.9
    nesterov: False
    weight_decay: 5e-05
)
SGD (
Parameter Group 0
    dampening: 0
    lr: 0.1
    momentum: 0.9
    nesterov: False
    weight_decay: 5e-05
)
SGD (
Parameter Group 0
    dampening: 0
    lr: 0.1
    momentum: 0.9
    nesterov: False
    weight_decay: 5e-05
)
SGD (
Parameter Group 0
    dampening: 0
    lr: 0.1
    momentum: 0.9
    nesterov: False
    weight_decay: 5e-05
)
SGD (
Parameter Group 0
    dampening: 0
    lr: 2.0796608879436464e-11
    momentum: 0.9
    nesterov: False
    weight_decay: 5e-05
)
SGD (
Parameter Group 0
    dampening: 0
    lr: 0.1
    momentum: 0.9
    nesterov: False
    weight_decay: 5e-05
)
SGD (
Parameter Group 0
    dampening: 0
    lr: 0.1
    momentum: 0.9
    nesterov: False
    weight_decay: 5e-05
)
/search/speech/yh/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Train Epoch     #1:   0%|          | 0/5005 [00:00<?, ?it/s]SGD (
Parameter Group 0
    dampening: 0
    lr: 0.1
    momentum: 0.9
    nesterov: False
    weight_decay: 5e-05
)

As we can see above, only one gpu's lr is ` lr: 2.0796608879436464e-11`, but others is still 0.1 which is the init lr.

but in your code , `    # Horovod: broadcast parameters & optimizer state.
    hvd.broadcast_parameters(model.state_dict(), root_rank=0)
    hvd.broadcast_optimizer_state(optimizer, root_rank=0)`   which says : Horovod: broadcast parameters & optimizer state.  So maybe this is a very serious bug???

Please confirm it as soon as possible. Thanks a lot! :)

**Comment by tgaddair** [comment]
Hey @yh673025667, thanks for sharing this, I'll definitely look into this.  But first, please give me a bit more info to help with repro, specifically:

- What version of PyTorch are you using?
- What version of Horovod are you using?
- What's the full `mpirun` command you're using?

**Comment by yh673025667** [comment]
@tgaddair Hi, some more details are as follows

- Pytorch:0.4.1

- Horovod:0.14.1

- `mpirun` command:`mpirun -np 32 \
	-H server1:8,server2:8,server3:8,server4:8 \
	-bind-to none -map-by slot \
	-x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH -x NCCL_IB_DISABLE=1 \
	-x NCCL_SOCKET_IFNAME=ib0 \
	-x UCX_MEM_MALLOC_HOOKS=no \
        -mca pml ob1 -mca btl ^openib \
	-mca btl_tcp_if_exclude lo,docker0,flannel.1,ib0 \
	python train_imagenet_distributed_horovod.py \
	--epochs 25 \
	--train-dir ~/data/ImageNet/train \
	--val-dir ~/data/ImageNet/val \
	--log-dir ./horovod_imagenet_v9 \
	--load-ckpt horovod_imagenet_v6/checkpoint-180.pth.tar \
	--warmup-epochs 0 \
	--worker 36 \`

here sever1 to server4 means my real machine ip, but i don't want to share public, thus i replace it to "server"

**Comment by tgaddair** [comment]
Hey @yh673025667, sorry for the late response.  This behavior was expected as we weren't broadcasting the optimizer "options" like learning rate, which are typically static unless changed from outside the optimizer.  However, #562 introduces a change that will address this issue if you'd like to check it out.
